src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartLegendConfiguration.tsx

```diff
@@ -168,7 +168,7 @@ const ChartLegendConfiguration = ({
       position: true,
       items: initialValues.items.map(item =>
         mapValues(item, () => true),
-      ) as FormikTouched<LegendItem>[],
+      ) as FormikTouched<LegendItem | unknown>[],
```

Comment:
Why unknown here?

Reply:
It's an attempt to fix the following error, any ideas on a better way?

```error TS2352: Conversion of type '{ dataSet: boolean; label: boolean; colour: boolean; symbol?: boolean | undefined; lineStyle?: boolean | undefined; inlinePosition?: boolean | undefined; }[]' to type 'FormikTouched<LegendItem>[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
src/pages/release/datablocks/components/chart/ChartLegendConfiguration.tsx(169,14): error TS2352: Conversion of type '{ dataSet: boolean; label: boolean; colour: boolean; symbol?: boolean | undefined; lineStyle?: boolean | undefined; inlinePosition?: boolean | undefined; }[]' to type 'FormikTouched<LegendItem>[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ dataSet: boolean; label: boolean; colour: boolean; symbol?: boolean | undefined; lineStyle?: boolean | undefined; inlinePosition?: boolean | undefined; }' is not comparable to type 'FormikTouched<LegendItem>'.
    Types of property 'dataSet' are incompatible.
      Type 'boolean' is not comparable to type 'FormikTouched<DataSet>'.

Reply:
That gives me the same error.

Reply:
Ohhh, it's because `mapValues` will only work on the direct properties of `LegendItem`. However, `LegendItem` also has a `dataSet` property, so to get a correct `FormikTouched` (which can be a deeply nested object with boolean values for each property), we'd also need to map over this as well.

One way to fix this could be the following:

```ts
return {
  position: true,
  items: initialValues.items.map(item => {
    return {
      ...mapValues(item, () => true),
      dataSet: mapValues(item.dataSet, () => true),
    };
  }),
};
```

However, we might be able to just omit the `dataSet` property completely i.e.

```
return {
  position: true,
  items: initialValues.items.map(item =>
    mapValues(omit(item, 'dataSet'), () => true),
  ),
};
```

I'm not sure if there are runtime issues with either approach however, so we'll probably want to test what actually happens if we do either.

Another approach would be the following:

```ts
return {
  position: true,
  items: initialValues.items.map(
    item =>
      mapValues(item, () => true) as FormikTouched<
        Omit<LegendItem, 'dataSet'>
      >,
  ),
};
```

However, I'm not sure if it passes compilation (I'm not on the branch!)

Reply:
Omiting dataSet seems to wot work fine so I've gone with that. Thanks for your help!